internal/controller/eventbus/eventlog/eventlog.go

```diff
@@ -396,20 +398,20 @@ func (mgr *eventlogManager) UpdateSegmentReplicas(ctx context.Context, leaderID
 	return nil
 }
 
-func (mgr *eventlogManager) GetSegmentByBlockID(block *metadata.Block) (*Segment, error) {
+func (mgr *eventlogManager) GetSegmentByBlockID(block *metadata.Block) (Segment, error) {
 	v, exist := mgr.eventLogMap.Load(block.EventlogID.Key())
 	if !exist {
-		return nil, errors.ErrEventLogNotFound
+		return Segment{}, errors.ErrEventLogNotFound
 	}
 	el, _ := v.(*eventlog)
-	return el.get(block.SegmentID), nil
+	return el.get(block.SegmentID).Copy(), nil
```

Comment:
need lock

Reply:
I can't get why the point-get operation needs a lock.

Reply:
> I can't get why the point-get operation needs a lock.

copy will access data of segment, and it is unsafe without lock.